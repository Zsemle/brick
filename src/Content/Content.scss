@import '../style/layout.scss';

.content {
  padding: $space-small;

  @media (min-width: $bp-medium) {
    padding: 1.6rem 0 5rem 0;
    min-height: calc(100vh - 18rem);
  }
}
