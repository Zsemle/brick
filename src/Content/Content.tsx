import React, { FunctionComponent } from 'react'
import './Content.scss'

interface ContentProps {
  children?: React.ReactNode
}

const Content:FunctionComponent = ({ children }:ContentProps):JSX.Element => {
  return (
    <main className="content">
      {children}
    </main>
  )
}

export default Content
