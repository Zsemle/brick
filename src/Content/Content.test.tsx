import { shallow, ShallowWrapper } from 'enzyme'
import React from 'react'
import Content from './Content'

describe('the content component', () => {
  it('renders the component', () => {
    const wrapper:ShallowWrapper = shallow(<Content />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the component with it\'s children', () => {
    const wrapper:ShallowWrapper = shallow(
      <Content>
        <div>{'test content 1'}</div>
        <div>{'test content 2'}</div>
      </Content>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
