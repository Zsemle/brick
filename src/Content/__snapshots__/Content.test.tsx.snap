// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the content component renders the component 1`] = `
<main
  className="content"
/>
`;

exports[`the content component renders the component with it's children 1`] = `
<main
  className="content"
>
  <div>
    test content 1
  </div>
  <div>
    test content 2
  </div>
</main>
`;
